vfio-ubind/src/main.rs

```diff
@@ -0,0 +1,112 @@
+#[macro_use] extern crate log;
+
+extern crate argparse;
+extern crate env_logger;
+extern crate regex;
+extern crate users;
+
+use std::path::Path;
+use regex::Regex;
+use argparse::{ArgumentParser, StoreTrue, Store};
+use std::fs::{OpenOptions, read_link};
+use std::io::prelude::*;
+use users::{get_user_by_uid, get_effective_uid, get_current_uid, get_group_by_name};
+use users::os::unix::GroupExt;
+
+fn pretty_write(path: &Path, content : &str, dryrun: bool) {
+	if dryrun {
+		println!("writing {} into {}", content, path.display());
+	} else { 
+		info!("writing {} into {}", content, path.display());
+	
+		let mut file = OpenOptions::new().write(true).open(&path).expect(&format!("Failed to open {}", path.display()));
```

Comment:
Not 100% equivalent:

>This function will create a file if it does not exist, and will truncate it if it does.

We don't really care about the truncation aspect but we definitely don't want to try to create random files in sysfs (not like it would work anyways but it doesn't make sense to even try).